modules/Installer/AWS.Tools.Installer.psm1

```diff
@@ -501,75 +463,127 @@ function Install-AWSToolsModule {
             else {
                 "AWS.Tools.$_"
             }
-        }
+        } | Sort-Object -Unique
 
-        if (($Name -like 'AWS.Tools.*' | Sort-Object -Unique).Count -ne $Name.Count) {
-            throw "The Name parameter must contain only AWS.Tools modules without duplicates."
+        if ($Name -notlike 'AWS.Tools.*') {
+            throw "The Name parameter must contain only AWS.Tools modules."
         }
 
         if ($Name -eq 'AWS.Tools.Installer') {
             throw "AWS.Tools.Installer cannot be used to install AWS.Tools.Installer. Use Update-Module instead."
         }
-        
-        if ($RequiredVersion) {
-            $versionToInstall = $RequiredVersion
+
+        [PSObject[]]$availableModulesToInstall = Find-AWSToolsModule -Name $Name -Proxy $Proxy -ProxyCredential $ProxyCredential
+
+        [System.Version]$availableVersion = [System.Version[]]$availableModulesToInstall.Version | Measure-Object -Minimum | Select-Object -Expand Minimum
+
+        if ($MinimumVersion -and $MinimumVersion -gt $availableVersion) {
+            throw "The maximum version available is $availableVersion."
         }
-        else {
-            $versionToInstall = Get-AvailableModuleVersion -Name $Name -MaximumVersion $MaximumVersion -Proxy $Proxy -ProxyCredential $ProxyCredential
-            Write-Verbose "[$($MyInvocation.MyCommand)] Installing AWS Tools version $versionToInstall"
+        if ($RequiredVersion -and $RequiredVersion -gt $availableVersion) {
+            throw "The maximum version available is $availableVersion."
+        }
+        if ($MinimumVersion -and $RequiredVersion -and $MinimumVersion -gt $RequiredVersion) {
+            throw 'Parameter MinimumVersion is greater than RequiredVersion.'
+        }
+        if ($MaximumVersion -and $RequiredVersion -and $MaximumVersion -lt $RequiredVersion) {
+            throw 'Parameter MaximumVersion is less than RequiredVersion.'
+        }
+        if ($MaximumVersion -and -not $RequiredVersion -and $MaximumVersion -lt $availableVersion) {
```

Comment:
How are these $null? The Get-CleanVersion Cmdlet returns 0 by default.

Reply:
`Get-CleanVersion` returns `$null` if its parameter is `$null`.